ceph/ceph.py

```diff
@@ -1412,72 +1457,52 @@ def exec_command(self, **kw):
                 check_ec: False will run the command and not wait for exit code
 
         """
+        if kw.get("long_running"):
+            return self.long_running(**kw)
 
-        if kw.get("sudo"):
-            ssh = self.rssh
-        else:
-            ssh = self.ssh
+        timeout = kw["timeout"] if kw.get("timeout") else 300
+        ssh = self.rssh() if kw.get("sudo") else self.ssh()
 
-        if kw.get("timeout"):
-            timeout = kw["timeout"]
-        else:
-            timeout = 120
+        logger.info(
+            f"Running command {kw['cmd']} on {self.ip_address} timeout {timeout}"
+        )
 
-        logger.info("Running command %s on %s", kw["cmd"], self.ip_address)
-        stdin = None
-        stdout = None
-        stderr = None
         if self.run_once:
             self.ssh_transport().set_keepalive(15)
             self.rssh_transport().set_keepalive(15)
 
-        if kw.get("long_running"):
-            logger.info("long running command --")
-            channel = ssh().get_transport().open_session()
-            channel.exec_command(kw["cmd"])
-            read = ""
-            while True:
-                if channel.exit_status_ready():
-                    ec = channel.recv_exit_status()
-                    break
-
-                rl, wl, xl = select([channel], [], [channel], 4200)
-
-                if len(rl) > 0:
-                    data = channel.recv(1024)
-                    data = data.decode(errors="ignore")
-                    logger.info(data)
-                    read += data
-
-                if len(xl) > 0:
-                    data = channel.recv(1024)
-                    data = data.decode(errors="ignore")
-                    logger.error(data)
-                    read += data
-
-            return read, ec
-
+        stdin = None
+        stdout = str()
+        stderr = str()
+        _stdout = None
+        _stderr = None
         try:
-            stdin, stdout, stderr = ssh().exec_command(kw["cmd"], timeout=timeout)
+            stdin, _stdout, _stderr = ssh.exec_command(kw["cmd"], timeout=timeout)
+            for line in _stdout:
+                if line:
+                    stdout += line
+            for line in _stderr:
+                if line:
+                    stderr += line
```

Comment:
Can we return the output as is? Here, we may skew the output... by appending as a single string

Reply:
need to see whether stream.read() works out or not.

Reply:
This channel.stream.read() working.
```
            stdout = _stdout.read().decode()
            stderr = _stderr.read().decode()
```